### ISSUE TITLE
Color-contrast errors out on CSS color spaces in text-shadow

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4080]
This results in an incomplete result from axe-core:

```html
<h2 style="color: #ccc; text-shadow: 
  3px 3px oklch(60% 0.57 181),
  0px 3px oklch(60% 0.57 181),
  0px -3px oklch(60% 0.57 181),
  3px 0px oklch(60% 0.57 181),
  -3px 0px oklch(60% 0.57 181),
  -3px -3px oklch(60% 0.57 181);
">Hello world</h2>
```

This is the error:

```
Error: Unable to process text-shadows: oklch(0.6 0.57 181) 3px 3px 0px, oklch(0.6 0.57 181) 0px 3px 0px, oklch(0.6 0.57 181) 0px -3px 0px, oklch(0.6 0.57 181) 3px 0px 0px, oklch(0.6 0.57 181) -3px 0px 0px, oklch(0.6 0.57 181) -3px -3px 0px
```

A little while back we fixed this problem with use of `color` and `background-color`, but it seems we missed something on `text-shadow`. Might be good to go over the rest of the color commons, see if we missed anything else too.

**Comment by padmavemulapati** [comment]
Validated using latest axe-core develop branch code base, 

for the above code:

```
<h2 style="color: #ccc; text-shadow: 
  3px 3px oklch(60% 0.57 181),
  0px 3px oklch(60% 0.57 181),
  0px -3px oklch(60% 0.57 181),
  3px 0px oklch(60% 0.57 181),
  -3px 0px oklch(60% 0.57 181),
  -3px -3px oklch(60% 0.57 181);
">Hello world</h2>

```

`color-contrast` rule is failing



https://github.com/dequelabs/axe-core/assets/47409579/a0e43fa5-e3b9-4972-a8e8-bf7b971dec03
